### ISSUE TITLE
regular Java document highlighting broken in VSCode when extension is active

### ISSUE THREAD

**Issue by martinlippert** [spring-projects/sts4#1380]
This is the follow-up from https://github.com/redhat-developer/vscode-java/issues/3787
Looks like regular document highlighting in Java source code is broken when the Spring Boot Tools extension is active.

**Comment by martinlippert** [comment]
Maybe a similar issue as we had with semantic tokens not being merged when coming from different language servers?

**Comment by martinlippert** [comment]
Closing this here already, since the fix seems to work nicely. Nevertheless, we should look into the VSCode side as well @BoykoAlex.

**Comment by BoykoAlex** [comment]
@martinlippert I agree, yes, will report to VSCode... If I'm to work on the PR I'd start with semantic tokens though. It's been reported a while ago, it has been idle for a long time and it affects us more as we have rather ugly workaround ;-)

**Comment by BoykoAlex** [comment]
Raised https://github.com/microsoft/vscode/issues/230946 but I would not say we are affected much by this issue... The range is the parameter. If range is inside the query unlikely Java would provide anything. If the cursor is at Java method parameter and we'd like to highlight that parameter in the query then this would be a problem... I feel semantic tokens not merged is far more painful then this...

**Comment by martinlippert** [comment]
@BoykoAlex Thanks for raising this with the VSCode team. And I agree that the fix that we have in place for now (returning null instead of an empty list) is good enough here. Once the VSCode piece got fixed, we can think about the special case that you mentioned.

And yes, getting the semantic tokens merged is far more important, totally agree.
